cli/run.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (C) 2020. Huawei Technologies Co., Ltd. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+import os
+import time
+import sys
+import signal
+import subprocess
+import webbrowser
+from contextlib import contextmanager
+
+import click
+
+
+@contextmanager
+def kill_process_group_afterwards():
+    os.setpgrp()
+    try:
+        yield
+    finally:
+        # Kill all processes in my group
+        os.killpg(0, signal.SIGKILL)
+
+
+@click.command(
+    name="run",
+    help="Run an experiment on a scenario",
+    context_settings=dict(ignore_unknown_options=True),
+)
+@click.option(
+    "--envision",
+    is_flag=True,
+    default=False,
+    help="Start up Envision server when running an experiment",
+)
+@click.argument(
+    "script_path", type=click.Path(exists=True), metavar="<script>", required=True
+)
+@click.argument("script_args", nargs=-1, type=click.UNPROCESSED)
+def run_experiment(envision, script_path, script_args):
+    with kill_process_group_afterwards():
+        if envision:
+            subprocess.Popen(
+                ["scl", "envision", "start", "-s", "./scenarios", "-p", "8081"],
+            )
+            # Just in case: give Envision a bit of time to warm up
+            time.sleep(2)
+            webbrowser.open_new_tab("http://localhost:8081")
+
+        script = subprocess.Popen(
+            [sys.executable, script_path, *script_args],
+        )
+        script.communicate()
```

Comment:
We're going to need to see `stdout` and `stderr` (and probably want to see it "live" / unbuffered).

Reply:
You do get the stdout and stderr live, when you run the command. They are getting directed to STDOUT by default not to PIPE.

Reply:
oops, sorry, yeah I see you're right!  cool.